nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,11 +439,217 @@ async fn run() -> Result<()> {
+    tracing::debug!("Finalizing classifications for event: {}", assign_event_id);
+    let mut answers: HashMap<String, i32> = HashMap::new();
+
+    // Let's first go through all tasks and construct a map of
+    // answer -> count
+    for task in tasks.iter() {
+        let identifier: String = match get_task_identifier(task) {
+            Some(val) => {
+                if val.is_empty() {
+                    tracing::debug!("Failed to get identifier");
+                    continue;
+                }
+                val
+            }
+
+            None => {
+                tracing::debug!("Failed to get identifier");
+                continue;
+            }
+        };
+        match answers.entry(identifier) {
+            Entry::Occupied(mut entry) => {
+                entry.insert(entry.get() + 1);
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(1);
+            }
+        }
+    }
+
+    // Let's check if there's an answer which majority of the tasks
+    // have, if we see multiple answers with same counts from all task results
+    // we retry and wait for more results
+    // max_count count number of results having same max count
+    let mut max_count = 0;
+    // max_answer is number of results having max answer
+    let mut max_answer = 0;
+    let mut answer = String::new();
+    for (result, count) in answers {
+        if count == max_answer {
+            max_count += 1;
+        } else if count > max_answer {
+            max_count = 1;
+            max_answer = count;
+            answer = result;
+        }
+    }
+    // This is when we are going to retry because there is not a single
+    // answer occurs more than any others
+    let mut accepted_result: Option<String> = None;
+    if max_count == 1 {
+        accepted_result = Some(answer);
+    }
+    let (result, winners) = get_winners(tasks, accepted_result);
+    // TODO(wangjun.hong): Replace below with a single update
+
+    if !winners.is_empty() && !result.is_none() {
+        reward(assign_event_id.clone(), &winners, result.unwrap(), classifier_result_collection,
+        collection, 1).await;
+    }
+
+    Ok(())
+}
+
+/**
+Go through all tasks and return all winners and the result payload
+When result is None, no winners are returned
+**/
+fn get_winners(
+    tasks: &Vec<AssignerTask>,
+    result: Option<String>
+) -> (Option<ResultPayload>, Vec<Sender>) {
+    tracing::debug!("Result is: {:?}", result);
+    let mut winners: Vec<Sender> = vec![];
+    let mut result_payload: Option<ResultPayload> = None;
+    tasks.iter().for_each(|task| {
+        let task_result = get_task_identifier(task);
+        if task_result == result && !task_result.is_none() {
+            winners.push(task.worker.clone());
+            result_payload = Some(task.result.clone());
+        }
+    });
+
+    (result_payload, winners)
+}
+
+// Return values from a task identifier, in case of
+// failure happened, return None
```

Comment:
I disagree: an error should be different from `None`. You can leave it as is but we are losing information here, but it's also making the code very verbose (FYI, there are ways to compact this).

All that said, please, remove the blank lines between every match branch, there are not needed